src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,202 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * TaskList that contains the list of Tasks and its operations.
+ *
+ * @author Aaron Pang
+ * @version CS2103T AY22/23 Sem 1
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int size;
+
+    /**
+     * Constructs the TaskList
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+        this.size = 0;
+    }
+
+    /**
+     * Constructs the TaskList with previous inputs
+     * @param dataList Previous input data
+     */
+    public TaskList(ArrayList<String> dataList) {
+        this.tasks = new ArrayList<>();
+        for (String data : dataList) {
+            tasks.add(Task.loadTask(data));
+        }
+        this.size = tasks.size();
+    }
+
+    /**
+     * Saves the TaskList after user input bye.
+     * @return ArrayList of the input that has been saved.
+     */
+    public ArrayList<String> bye() {
+        return saveTasks();
+    }
+
+    /**
+     * Saves the TaskList
+     * @return ArrayList of the input that has been saved.
+     */
+    public ArrayList<String> saveTasks() {
+        ArrayList<String> list = new ArrayList<>();
+        for (Task task : tasks) {
+            list.add(task.saveTask());
+        }
+        return list;
+    }
+
+    /**
+     * Marks a task.
+     *
+     * @param in Input of the user.
+     */
+    public void markTask(String in) {
+        char n = in.charAt(5);
+        int number = Character.getNumericValue(n) - 1;
+        Task t = tasks.get(number);
+        t.markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(t.toString());
```

Comment:
I believe that the lines in this code have different level of abstraction. Perhaps you can create methods that does things like calculate the task index?